### PR TITLE
Add a deprecation notices section

### PR DESCRIPTION
**Pull request by ddtmachado**
### What does this PR do?

This new section should hold information about:

* active supported branches
* our rules on support and versioning scheme
* deprecation notices for features (none included in the PR)

It also removes v1 doc links from readme as I believe they should not be there anymore.

### Motivation

Clearer communication

### More

- [ ] Added/updated tests
- [X] Added/updated documentation

### Additional Notes

A separate page should be created to map planned feature deprecation and any changes in default behavior that can affect users while upgrading. One good example is the fix for the Kubernetes cross-namespace references, that unintentionally broke configuration for users relying on the bugged behavior.

### CODE REVIEW

File: docs/content/deprecation/releases.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+| 2.4     | Jan 19, 2021 | Ended Aug 17, 2021 |       No         |
+| 2.3     | Sep 23, 2020 | Ended Jan 19, 2021 |       No         |
+| 2.2     | Mar 25, 2020 | Ended Sep 23, 2020 |       No         |
+| 2.1     | Dec 11, 2019 | Ended Mar 25, 2020 |       No         |
+| 2.0     | Sep 16, 2019 | Ended Dec 11, 2019 |       No         |
+| 1.7     | Sep 24, 2018 | Ended Dec 31, 2021 |  Contact Support |
+
+??? example "Active Support / Security Support"
+
+    **Active support**: represents a version that still receives small enchancements to existing features as well as bug fixes. <br>
```
Comment (tomMoulard) [inline docs/content/deprecation/releases.md:20]:
```suggestion
    **Active support**: represents a version that still receives small enhancements to existing features as well as bug fixes.  
```

Comment (tomMoulard) [reply docs/content/deprecation/releases.md:20]: Two spaces at the end of line is equivalent to `<br>` in Markdown.

File: docs/content/deprecation/releases.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    **Security support**: represents a version that receives critical bug and security fixes.
+
+This page is maintained and updated periodically to reflect our roadmap and any decisions affecting the end of support for Traefik Proxy.
+
+Please refer to our migration guides for specific instructions on upgrading between versions, an example is the [v1 to v2 migration guide](../migration/v1-to-v2.md).
+
+!!! important "Note:"
+    All target dates on end of support or feature removal announcements may be subject to change.
```
Comment (tomMoulard) [inline docs/content/deprecation/releases.md:28]:
```suggestion
!!! important "All target dates for end of support or feature removal announcements may be subject to change."
```

File: docs/content/deprecation/releases.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Releases
+
+## Versions
+
+Below is non-exhaustive list of versions and their maintenance status:
```
Comment (kevinpollet) [inline docs/content/deprecation/releases.md:5]:
```suggestion
Below is a non-exhaustive list of versions and their maintenance status:
```

File: docs/content/deprecation/releases.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+This page is maintained and updated periodically to reflect our roadmap and any decisions affecting the end of support for Traefik Proxy.
+
+Please refer to our migration guides for specific instructions on upgrading between versions, an example is the [v1 to v2 migration guide](../migration/v1-to-v2.md).
+
+!!! important "All target dates for end of support or feature removal announcements may be subject to change."
+
+## Versioning Scheme
+
+The Traefik Proxy project follows the [semantic versioning](https://semver.org/) scheme and maintains a separate branch for each minor version, the main branch always represent the next upcoming minor or major version.
```
Comment (kevinpollet) [inline docs/content/deprecation/releases.md:31]:
```suggestion
The Traefik Proxy project follows the [semantic versioning](https://semver.org/) scheme and maintains a separate branch for each minor version. The main branch always represents the next upcoming minor or major version.
```

**Review by kevinpollet — APPROVED** [review]
LGTM

**Pull request merged**